### ISSUE TITLE
composer.json editor adds unintended changes to file when opened

### ISSUE THREAD

**Issue by o-alquimista** [eclipse/pdt#107]
**Describe the bug**
The composer.json file editor is unexpectedly reformatting its source code every time you open it. The first thing you notice is the asterisk `*` indicating that the file has unsaved changes. You cannot undo these changes.

**Describe the eclipse environment**
Eclipse IDE for PHP Developers (includes Incubating components)
Version: 2021-03 (4.19.0)
Build id: 20210312-0638

Installed with the Eclipse installer downloaded from eclipse.org.

**Describe your system**
 - OS: Ubuntu 20.04 LTS and Fedora Workstation 34 (my desktop and my laptop)
 - PHP Version: 7.4 (Ubuntu) and 8.0 (Fedora)
 - Webserver Version: httpd 2.4
 - Composer: 2.0.13

**To Reproduce**
1. Open a composer.json file and switch from the "Overview" to the "composer.json" tab
2. Now that the file has unsaved changes (indicated by the asterisk `*`), you can see that spaces have been replaced with tabs, and there's a space character preceding every colon `:`. You cannot undo these with `Ctrl + Z`.
3. Without making any changes to the file, switch in and out of the "composer.json" tab, and you'll see that the asterisk `*` is no longer there to indicate unsaved changes. However, the file is not the same any more, as if the previously unsaved changes were automatically saved and committed.
4. But Eclipse is wrong! The actual file on disk is not even touched. You can check that by opening composer.json with another editor. Phew!

**Expected behavior**
The file should never be changed just by opening it.

**Additional context**
Related bug reports:
- [Composer.json editor page change file after opening](https://bugs.eclipse.org/bugs/show_bug.cgi?id=501977)
- [composer.json is not saved when edited](https://bugs.eclipse.org/bugs/show_bug.cgi?id=563691)

**Comment by pounard** [comment]
Yes I frequently experience this bug as well, althought I usually use the généraux JSON editor for editing composer files.

**Comment by spider4216** [comment]
The same issue and I use "Generic Text Editor" to avoid that behavior
